workflow/scripts/build_egon_data.py

```diff
@@ -0,0 +1,71 @@
+# -*- coding: utf-8 -*-
+import logging
+
+logger = logging.getLogger(__name__)
+import json
+
+import geopandas as gpd
+import pandas as pd
+
+"""
+    Load and prepares the egon data about district heating in Germany on NUTS3
+    level.
+
+    Output:
+        GeoDataFrame: A GeoDataFrame containing the processed egon data.
+"""
+
+
+if __name__ == "__main__":
+    if "snakemake" not in globals():
+        import os
+        import sys
+
+        os.chdir(os.path.dirname(os.path.abspath(__file__)))
+
+        path = "../submodules/pypsa-eur/scripts"
+        sys.path.insert(0, os.path.abspath(path))
+        from _helpers import mock_snakemake
+
+        snakemake = mock_snakemake(
+            "build_egon_data",
+            run="KN2045_Bal_v4",
+        )
+
+logger.info("Retrieving and cleaning egon data")
+
+nuts3 = gpd.read_file(snakemake.input.nuts3)[
+    ["index", "pop", "geometry"]
+]  # Keep only necessary columns
+
+internal_id = {
```

Comment:
does this need to be hard-coded or can this be read from the meta data as well?

Reply:
I am not sure, but I think the nuts3_shapes.geojson is created in `build_shapes` and only contains the usual metadata fields like "name", "type", and "crs". Did you have anything else in mind?

Reply:
This comment was about the `internal_id`dictionary. I think on the demand regio website they also have a `metadata.json` from which you can read in the mapping from number -> tech. But it is fine, if you cannot find it, we can leave it as it is.

Reply:
Using the metadata now.